extra/interface/src/main/java/org/spongepowered/configurate/interfaces/meta/defaults/DefaultBoolean.java

```diff
@@ -0,0 +1,28 @@
+package org.spongepowered.configurate.interfaces.meta.defaults;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation provides a default value for the annotated method.
+ * Because of annotation limits, there is an annotation for:
+ * {@link DefaultBoolean booleans}, {@link DefaultDecimal decimals},
+ * {@link DefaultNumeric numerics} and {@link DefaultString Strings}.
+ *
+ * @since 4.2.0
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.METHOD, ElementType.FIELD})
+public @interface DefaultBoolean {
```

Comment:
I feel like annotations are too limited to properly express defaults -- maybe it'd make sense to do something likes immutables' `defaultIsDefault` option (or a similar per-method annotation), which would source the default value from the `default` implementation of the interface.

Reply:
I was planning to add that, but instead of adding an annotation to it it'd automatically use the default value.
The idea behind it is that in a config class/interface you expect config nodes, and if there are methods that aren't related to config nodes you `@Exclude` them. Does that also fit your vision?